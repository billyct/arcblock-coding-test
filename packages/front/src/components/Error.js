import {Link} from 'wouter'

const Error = ({children}) => (
  <div className='text-center my-10'>
    <div className='text-4xl font-bold'>
      Oops!
    </div>
    <div className='mt-10'>
      {children}
    </div>
    <div className='text-sm mt-10'>
      There are some links maybe helpful:
      <Link href='/'>
        <a className='ml-2 text-blue-500'>Explorer</a>
      </Link>
    </div>
  </div>
)

export default Error
