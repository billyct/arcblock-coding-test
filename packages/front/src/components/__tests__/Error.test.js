import {render} from '@testing-library/react'

import Error from '../Error'

test('Error Component', () => {
  const {container} = render(
    <Error>
      an error message
    </Error>
  )

  expect(container).toMatchSnapshot()
})
