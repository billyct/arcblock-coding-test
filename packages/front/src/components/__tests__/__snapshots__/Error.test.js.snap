// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component 1`] = `
<div>
  <div
    class="text-center my-10"
  >
    <div
      class="text-4xl font-bold"
    >
      Oops!
    </div>
    <div
      class="mt-10"
    >
      an error message
    </div>
    <div
      class="text-sm mt-10"
    >
      There are some links maybe helpful:
      <a
        class="ml-2 text-blue-500"
        href="/"
      >
        Explorer
      </a>
    </div>
  </div>
</div>
`;
